# Share Price Of Mahindra and Mahindra 


Several factors, including economic conditions, political stability, global economic trends, and investor sentiment, contribute to India's stock market volatility.
The stock market's volatility is largely determined by the state of the economy. The stock market can be significantly affected by fiscal policies, inflation, interest rates, GDP growth, and other factors. For example, high interest rates or inflation can make it harder for consumers to buy things, which reduces demand for goods and services and ultimately reduces profits for businesses, which has an effect on stock prices. In a similar vein, stock prices can rise as a result of an increase in investor confidence and a positive GDP growth rate.
The volatility of the stock market is also influenced by political stability. Investor confidence can be impacted by political instability, such as frequent policy shifts or corruption, which can cause stock prices to fall. On the other hand, stock prices can rise as a result of an improvement in the political climate and an increase in investor confidence.
India's stock market volatility is also significantly influenced by global economic trends. The Indian stock market can be affected in a big way by changes in major economies like the US, China, and Europe. For instance, the Indian stock market was significantly affected by the 2008 global economic recession, which resulted in a sharp decline in stock prices.
Volatility in the stock market can also be influenced by investor sentiments. News events, earnings reports from companies, and market rumors all have an impact on investor sentiment. Stock prices can rise as a result of bullish market sentiment caused by positive news events or strong earnings reports. On the other hand, negative news occasions or powerless profit reports can prompt a negative market feeling, prompting a decrease in stock costs.

To summarize, investor sentiments, economic conditions, political stability, global economic trends, and the complexity of the Indian stock market all play a role in its volatility. To make well-informed decisions regarding investments, one must have a comprehensive understanding of these factors.

In [6]:
import pandas as pd
import numpy as np

In [7]:
import statsmodels.api as sm

In [13]:
#import the CSV

mnm = pd.read_csv('Mahindra.csv')

In [14]:
mnm.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2484 entries, 0 to 2483
Data columns (total 10 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Date                      2484 non-null   float64
 1   Open                      2484 non-null   float64
 2   High                      2484 non-null   float64
 3   Low                       2484 non-null   float64
 4   Close                     2484 non-null   float64
 5   Volume                    2484 non-null   float64
 6   No of Trades              2484 non-null   float64
 7   Net Turnover In Thousand  2484 non-null   float64
 8   Market Cap                2484 non-null   float64
 9   Sensex                    2484 non-null   float64
dtypes: float64(10)
memory usage: 194.2 KB


In [15]:
mnm.dtypes



Date                        float64
Open                        float64
High                        float64
Low                         float64
Close                       float64
Volume                      float64
No of Trades                float64
Net Turnover In Thousand    float64
Market Cap                  float64
Sensex                      float64
dtype: object

In [16]:
import matplotlib.pyplot as plt
import seaborn as sns

In [18]:
linear_model = sm.OLS(v,mnm['Sensex']).fit()

In [20]:
linear_model.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                                 OLS Regression Results                                
=======================================================================================
Dep. Variable:                  Close   R-squared (uncentered):                   0.957
Model:                            OLS   Adj. R-squared (uncentered):              0.957
Method:                 Least Squares   F-statistic:                          5.577e+04
Date:                Fri, 21 Apr 2023   Prob (F-statistic):                        0.00
Time:                        14:56:39   Log-Likelihood:                         -16007.
No. Observations:                2484   AIC:                                  3.202e+04
Df Residuals:                    2483   BIC:                                  3.202e+04
Df Model:                           1                                                  
Covariance Type:            nonrobust                                                  
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Sensex         0.0188   7.95e-05    236.151      0.000       0.019       0.019
==============================================================================
Omnibus:                      342.908   Durbin-Watson:                   0.005
Prob(Omnibus):                  0.000   Jarque-Bera (JB):              267.558
Skew:                          -0.708   Prob(JB):                     7.95e-59
Kurtosis:                       2.240   Cond. No.                         1.00
==============================================================================

Notes:
[1] R² is computed without centering (uncentered) since the model does not contain a constant.
[2] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

### The above method is wrong as evident from the lack of constant term in model

In [22]:
#Right way to do it

#define y and x and use them

x = mnm['Sensex']
y = mnm['Close']

x = sm.add_constant(x)

In [23]:
linear_model = sm.OLS(y,x).fit()

In [24]:
linear_model.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                  Close   R-squared:                       0.601
Model:                            OLS   Adj. R-squared:                  0.601
Method:                 Least Squares   F-statistic:                     3739.
Date:                Fri, 21 Apr 2023   Prob (F-statistic):               0.00
Time:                        15:00:06   Log-Likelihood:                -15656.
No. Observations:                2484   AIC:                         3.132e+04
Df Residuals:                    2482   BIC:                         3.133e+04
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const        233.0134      8.189     28.456      0.000     216.956     249.071
Sensex         0.0130      0.000     61.146      0.000       0.013       0.013
==============================================================================
Omnibus:                        8.354   Durbin-Watson:                   0.007
Prob(Omnibus):                  0.015   Jarque-Bera (JB):                7.890
Skew:                           0.104   Prob(JB):                       0.0193
Kurtosis:                       2.818   Cond. No.                     1.19e+05
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 1.19e+05. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

### 60% of the movement in price of the stock is explained by the movement in broad markets (as seen using Sensex)

## Descriptive Statistics
R-squared simply indicates how well a regression model fits the data. A perfect fit is indicated by an R-squared value of 1, which indicates that the independent variable can account for all of the variation in the dependent variable. In contrast, an R-squared value of 0 indicates that the independent variable's variation cannot account for all of the variation in the dependent variable.

In finance, R-squared is many times used to quantify the connection between stock costs and market files like the Sensex in India. A stock's performance is strongly correlated with the market's performance if the R-squared value between the stock and the Sensex is high. On the other hand, a low R-squared value indicates that the stock's performance is not closely linked to the market as a whole.

Let's take Mahindra, a company on the Sensex, as an example. As a result, we have compiled information regarding the Sensex and Mahindra stock's daily closing prices over time. We can then utilize relapse investigation to figure out the connection between Mahindra's stock cost and the Sensex.

Let's say we conduct a regression analysis and determine that the correlation between the Sensex and Mahindra's stock price is 0.601. This indicates that the Sensex's fluctuation can account for 60.1% of the variation in Mahindra's stock price. In other words, there is a moderate correlation between the performance of the market as a whole and the price of Mahindra's stock. However, the Sensex fluctuation does not account for all of the other factors that affect Mahindra's stock price.

Let's now contrast this with the Sensex's R-squared value. Let's say we conduct a regression analysis on the Sensex's daily closing prices and come up with a value of 0.957 for the R-squared. This indicates that the factors that have an effect on the market as a whole account for 95.7 percent of the Sensex's variation. To put it another way, the Sensex's performance is significantly influenced by the market as a whole.

We can learn a lot about the nature of the connection between individual stocks and the market as a whole from the difference in the R-squared values of Mahindra's stock and the Sensex. The overall market has a much stronger impact on the performance of the market as a whole than the performance of individual stocks, which is influenced by numerous factors, including company-specific factors like earnings and management decisions.

In practice, investors make investment decisions based on R-squared values. For instance, a financial backer who is keen on putting resources into Mahindra's stock might take a gander at the R-squared worth to decide how intently the stock's exhibition is attached to the presentation of the general market. If the R-squared value is high, an investor who believes the market as a whole will perform well may conclude that Mahindra's stock is a good investment. Then again, if the R-squared esteem is low, the financial backer might infer that Mahindra's stock is a hazardous speculation, as the stock's presentation might be impacted by factors other than the general market.